src/listener/builtin.ts

```diff
@@ -0,0 +1,41 @@
+import { InstallationStore, InstallationQuery } from "@slack/oauth";
+import { Logger } from '@slack/logger';
+import { Context } from "../types";
+
+export class TokenRevocationListeners {
+  private installationStore: InstallationStore;
+  private logger: Logger;
+
+  public constructor(installationStore: InstallationStore, logger: Logger) {
+    this.installationStore = installationStore;
+    this.logger = logger;
+  }
+
+  public handleTokensRevokedEvents(context: Context) {
+    const isEnterpriseInstall = context.isEnterpriseInstall;
+  
+    const installQuery: InstallationQuery<typeof isEnterpriseInstall> = {
+      isEnterpriseInstall: isEnterpriseInstall,
+      teamId: isEnterpriseInstall ? context.teamId : undefined,
+      enterpriseId: isEnterpriseInstall ? context.enterpriseId : undefined
+    };
+
+    this.installationStore.deleteInstallation(installQuery, this.logger);
+
+    // add logic to delete bot?
```

Comment:
There are two approaches:
1. Aligning with bolt-java/python
2. Opting for a simpler but more inflexible approach, which differs from bolt-python/java

**[Aligning with bolt-java/python]**

Aligning with bolt-python/java should be the safest, most clear, and flexible approach:
1. deleteInstallation with enterprise/team_id + user_id -> delete only user tokens
2. deleteBot with enterprise/team_id -> delete only bot tokens
3. deleteAll with enterprise/team_id -> delete both bot and user tokens

Review https://github.com/slackapi/bolt-python/blob/v1.21.2/slack_bolt/listener/builtins.py to better understand why bolt-python/java have three methods.

I don't believe this can cause any confusion. If a developer has already implemented the deleteInstallation method, bolt-js never calls the method without a user_id, so adding built-in support should not break anything. 

When we release this built-in listener support, we should mention the necessity of two more method implementation, though.

**[Opting for a simpler but more inflexible approach]**

If we try to only use deleteInstallation, the method can function in the following ways:
1. deleteInstallation with enterprise/team_id + user_id -> Deletes only user tokens
2. deleteInstallation with enterprise/team_id -> Deletes both bot and user tokens, similar to deleteAll

This approach should also work, but the downside is that when an app does not want to delete user tokens while deleting a bot token, deleteInstallation with enterprise/team_id does not behave this way.

While Bolt supports user-token-only use cases, we generally recommend having a bot token to enable many functionalities. Deleting a bot token usually means the app is no longer being used in the workspace. So, deleting all tokens when a bot token is revoked makes sense to many apps. With that being said, this inflexibility may limit developers options. Also, our platform intentionally provides two event types to differentiate the scenarios a single token revocation vs app uninstallation.

I recommend **[Aligning with bolt-java/python]**, but am open to any alternatives including the second option here. @filmaj do you have any option on this?

Reply:
> When we release this built-in listener support, we should mention the necessity of two more method implementation, though.

@seratch Personally I like the first option! Does this mean to proceed with the first option, we'll need to release a new `@slack/oauth` version with the [interface](https://github.com/slackapi/node-slack-sdk/blob/main/packages/oauth/src/installation-stores/interface.ts) for InstallationStore updated to include a deleteBot and deleteAll optional method, similar to [the Python implementation](https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/oauth/installation_store/installation_store.py)?

Reply:
> I recommend [Aligning with bolt-java/python]

Agreed 👍

Reply:
@hello-ashleyintech yes, you're right!